### PR TITLE
feat: add precompiled contracts for Greenfield link

### PR DESCRIPTION
**Pull request by yutianwu**
### Description

Add precompiled contracts for Greenfield link. The added precompiled contracts are active after `Fermat` hard fork.

### Rationale

By connecting Greenfield to opBNB, the dapp becomes more cost-effective and faster. 

### Changes

The greenfield-related precompile contracts should be added to opBNB and active after `Fermat` hard fork.

The below precompile contracts should be added:
```
common.BytesToAddress([]byte{102}): &blsSignatureVerify{}
common.BytesToAddress([]byte{103}): &cometBFTLightBlockValidate{}
```
The contract blsSignatureVerify is used to verify the aggregated BLS signatures from the relayers.
The contract cometBFTLightBlockValidate is used to validate the light blocks of cometBFT.

For more details about the two contracts, you can refer to [BEP 221](https://github.com/bnb-chain/BEPs/blob/master/BEPs/BEP221.md) and [BEP 126](https://github.com/bnb-chain/BEPs/blob/master/BEPs/BEP126.md)

### CODE REVIEW

File: core/vm/lightclient/lightclient.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+			return false, err
+		}
+	}
+
+	// update consensus state
+	cs.Height = uint64(block.Height)
+	cs.NextValidatorSetHash = block.NextValidatorsHash
+	cs.ValidatorSet = block.ValidatorSet
+
+	return !(bytes.Equal(cs.ValidatorSet.Hash(), block.ValidatorsHash)), nil
```
Comment (KeefeL) [inline core/vm/lightclient/lightclient.go:125]: This is not correct, please check the newest implementation, there is a bug here.
https://github.com/bnb-chain/bsc/blob/master/core/vm/lightclient/v2/lightclient.go#L121

File: core/vm/lightclient/lightclient.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
+// Package v2 is used for tendermint v0.34.22 and its compatible version.
```
Comment (KeefeL) [inline core/vm/lightclient/lightclient.go:1]: Package v2 -> Package lightclient?

File: Dockerfile [hunk 3]
```diff
@@ -6,6 +6,7 @@ ARG BUILDNUM=""
 # Build Geth in a stock Go builder container
 FROM golang:1.20-alpine as builder
 
+RUN apk add --no-cache build-base libc-dev
```
Comment (owen-reorg) [inline Dockerfile:9]: why not add to the existing line?

Comment (owen-reorg) [inline Dockerfile:9]: why not add to the existing line?

**Review by zjubfd — APPROVED** [review]

**Comment by zjubfd** [comment]
The precompile contract looks good to me, as I compare it to the ones on BSC.

**Review by redhdx — APPROVED** [review]

**Pull request merged**